Mathlib/Topology/Order/Basic.lean

```diff
@@ -46,6 +46,9 @@ of the discrete spaces `ℕ` or `ℤ`, or `ℝ` that could inherit a topology as
 and is in general not defeq to the one generated by the intervals. We make it available as a
 definition `Preorder.topology α` though, that can be registered as an instance when necessary, or
 for specific types.
+
+The order topology is only really useful on linear orders. To mirror this, all the `OrderTopology`
+API is only accessible when assuming `LinearOrder`, even when it could be generalised to `Preorder`.
```

Comment:
Please expand this a bit. A link to the zulip discussion would already help a bit.

But I think it would be even better if somewhere (here or on zulip and then link to it from here) there is a more detailed example of how the version with `Preorder` makes your life harder.